docs/jumpstart/week-6.md

```diff
@@ -0,0 +1,697 @@
+# Week 5: Advanced Backend and Testing
+
+## Agenda
+- Backend Basics Cont.
+- Testing
+  - JUnit 5
+  - Mockito
+- Advanced Backend
+  - DTOs in General
+  - Unmarshalling objects into DTOs
+  - Backend Validation
+  - Other Request Types
+  
+## Basic Plan
+1. Talk about testing in the software development world
+  - Types of tests
+  - Basic testing with JUnit
+    - Test Class Locations
+    - Show basic test examples
+    - `BeforeEach`/`BeforeAll` annotations
+    - Parameterized tests
+    - Exceptions
+  - Mocking objects with Mockito
+    - Mocking behavior and returns
+    - Verification
+    - Argument Captors
+    - `Externals` Class
+    - Show the test created for the backend api module
+    
+2. New Request Types
+  - `POST`, `PUT`, `PATCH`, and `DELETE`: what are they commonly used for
+  
+3. DTOs and Marshalling/Unmarshalling
+  - What are DTOs
+    - What are they used for
+  - Marshalling/Unmarshalling
+  - Validating unmarshalled objects/why it's necessary
+ 
+## Testing
+
+### Types of Tests
+
+You may be able to tell from the name that testing is a method used to help prove that your code works the way you expect 
+it to. If you haven't spent a lot of time working on tests before, whether that's because you're new to development or 
+never thought 
+that tests were important enough to spend time on, just know that they're extremely important in helping you 
+diagnose issues in your code (if you write meaningful tests) before releases or after making significant changes. 
+By the way, if you've heard of check-expects from fundies, they're basically the same thing! For smaller projects, tests 
+have less of an impact, even if they're still helpful. As your project grows and begins incorporating work from more 
```

Comment:
This sentence is kind of awkward, "For smaller projects, tests have less of an impact, even if they're still helpful." I might replace the even with but because test are still helpful

Reply:
Fixed.